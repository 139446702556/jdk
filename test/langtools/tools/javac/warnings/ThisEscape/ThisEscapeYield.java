/*
 * @test /nodynamiccopyright/
 * @bug 8015831
 * @compile/ref=ThisEscapeYield.out -Xlint:this-escape -XDrawDiagnostics ThisEscapeYield.java
 * @summary Verify 'this' escape detection handles leaks via switch expression yields
 */

public class ThisEscapeYield {

    public ThisEscapeYield(int x) {
        ThisEscapeYield y = switch (x) {
            case 3:
                if (x > 17)
                    yield this;
                else
                    yield null;
            default:
                yield null;
        };
        if (y != null)
            y.mightLeak();
    }

    public void mightLeak() {
    }
}
