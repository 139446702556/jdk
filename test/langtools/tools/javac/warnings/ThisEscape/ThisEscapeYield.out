ThisEscapeYield.java:21:24: compiler.warn.possible.this.escape
1 warning
